### PR TITLE
Mormalizing #37

### PR DESCRIPTION
**Pull request by TomHrm**

### CODE REVIEW

File: Backend/Solver/CrankNicolson/Iterate.c [hunk 1]
```diff
@@ -9,6 +10,19 @@
 // Speed of light in pm/as
 #define c 302
 
+static Complex **normalize(Complex **res, int n) {
+    for (int i = 0; i < n; i++) {
```
Comment (MetaColon) [inline Backend/Solver/CrankNicolson/Iterate.c:14]: Wrong brace style

File: Backend/Solver/CrankNicolson/Iterate.c [hunk 2]
```diff
@@ -9,6 +10,19 @@
...
+        double result = 0;
```
Comment (MetaColon) [inline Backend/Solver/CrankNicolson/Iterate.c:15]: `result` is a misleading variable name

File: Backend/Solver/CrankNicolson/Iterate.c [hunk 3]
```diff
@@ -67,6 +81,7 @@ Complex **Iterate1d(Function potential, Function psi0, int n)
         printf("i=%d\n", i);
 
         res[i] = matrixToArray(psi);
+
```
Comment (MetaColon) [inline Backend/Solver/CrankNicolson/Iterate.c:84]: Why do we need this blank line?

File: Backend/Solver/CrankNicolson/Iterate.c [hunk 4]
```diff
@@ -9,6 +10,19 @@
 // Speed of light in pm/as
 #define c 302
 
+static Complex **normalize(Complex **res, int n) {
```
Comment (MetaColon) [inline Backend/Solver/CrankNicolson/Iterate.c:13]: Making this method return `res` implies that it is immutable (which it isn't)

File: Backend/Solver/CrankNicolson/Iterate.c [hunk 5]
```diff
@@ -101,7 +116,8 @@ Complex **Iterate1d(Function potential, Function psi0, int n)
             potentialMatrix = createPotentialMatrix(potentialValues);
         }
     }
-    res[n-1] = matrixToArray(psi);
+    res[n - 1] = matrixToArray(psi);
+    res = normalize(res, n);
```
Comment (MetaColon) [inline Backend/Solver/CrankNicolson/Iterate.c:120]: The assignment to `res` is redundant, as the parameter is already altered.

**Review by MetaColon — CHANGES_REQUESTED** [review]

File: Backend/Solver/CrankNicolson/Iterate.c [hunk 6]
```diff
@@ -100,8 +114,9 @@ Complex **Iterate1d(Function potential, Function psi0, int n)
             // nx * nx
             potentialMatrix = createPotentialMatrix(potentialValues);
         }
+        res = normalize(res, n);
```
Comment (MetaColon) [inline Backend/Solver/CrankNicolson/Iterate.c:117]: As you normalize _all_ values, you only need to call `normalize` once (e.g. at the end)

**Review by MetaColon — CHANGES_REQUESTED** [review]

File: Backend/Solver/CrankNicolson/Iterate.c (hunk 5 above)
Comment (MetaColon) [reply Backend/Solver/CrankNicolson/Iterate.c]: You didn't adress this.

File: Backend/Solver/CrankNicolson/Iterate.c (hunk 4 above)
Comment (MetaColon) [reply Backend/Solver/CrankNicolson/Iterate.c]: You didn't adress this.

Comment (TomHrm) [reply Backend/Solver/CrankNicolson/Iterate.c]: which return type is appropriate?

Comment (MetaColon) [reply Backend/Solver/CrankNicolson/Iterate.c]: `void`

File: Backend/Solver/CrankNicolson/Iterate.c [hunk 7]
```diff
@@ -9,6 +10,22 @@
 // Speed of light in pm/as
 #define c 302
 
+void normalize(Complex **normalized, int n)
```
Comment (MetaColon) [inline Backend/Solver/CrankNicolson/Iterate.c:13]: This parameter name is misleading, as it isn't normalized by the time the method is called

File: Backend/Solver/CrankNicolson/Iterate.c [hunk 8]
```diff
@@ -101,7 +117,8 @@ Complex **Iterate1d(Function potential, Function psi0, int n)
             potentialMatrix = createPotentialMatrix(potentialValues);
         }
     }
-    res[n-1] = matrixToArray(psi);
+    normalize(res, n);
```
Comment (MetaColon) [inline Backend/Solver/CrankNicolson/Iterate.c:120]: The normalization should be done _after_ the last value is written to `res`

**Review by MetaColon — CHANGES_REQUESTED** [review]

**Review by MetaColon — APPROVED** [review]

**Pull request merged**